features/bootstrap/ProfileContext.php

```diff
@@ -17,10 +22,41 @@ public function enableFeature()
     }
 
     /**
-     * @Given /^([A-Za-z\s]+) has (\d+) annotations?$/
+     * @Given /^I am ([A-Za-z\s]+)$/
      */
-    public function profileHasAnnotations(string $name, int $number)
+    public function iAm(string $name)
     {
+        $this->name = $name;
+    }
+
+    /**
+     * @Given /^I have logged in$/
+     */
+    public function iHaveLoggedIn()
+    {
+        $session = $this->kernel->getContainer()->get('session');
+
+        $id = $this->createId($this->name);
+
+        $token = new PostAuthenticationGuardToken(new OAuthUser($id, $roles = ['ROLE_USER', 'ROLE_OAUTH_USER']), 'main', $roles);
+
+        $session->set('_security_main', serialize($token));
+        $session->save();
+
+        $cookie = new Cookie($session->getName(), $session->getId());
+        $this->getMink()->getSession()->getDriver()->getClient()->getCookieJar()->set($cookie);
+    }
+
+    /**
+     * @Given /^([A-Za-z\s]+) has (\d+) public( and private)? annotations?$/
+     * @Given /^(I) have (\d+) public( and private)? annotations?$/
```

Comment:
This has become a little messy putting them together, could separate.

Reply:
:+1: it's the only part that is difficult to follow